sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -784,8 +785,28 @@ case class GpuCast(
               convertTimestampOrNull(sanitizedInput, TIMESTAMP_REGEX_YYYY, "%Y"))))
 
       // handle special dates like "epoch", "now", etc.
-      specialDates.foldLeft(converted)((prev, specialDate) =>
+      val finalResult = specialDates.foldLeft(converted)((prev, specialDate) =>
         specialTimestampOr(sanitizedInput, specialDate._1, specialDate._2, prev))
+
+      // When ANSI mode is enabled, we need to throw an exception if any values could not be
+      // converted
+      if (ansiMode) {
+        val wasNotNull = withResource(input.isNull()) { wasNull =>
+          wasNull.not()
+        }
+        withResource(wasNotNull) { wasNotNull =>
+          withResource(finalResult.isNull) { isNull =>
+            withResource(wasNotNull.and(isNull)) { notConverted =>
+              if (notConverted.any().getBoolean) {
+                throw new DateTimeException(
```

Comment:
Sorry I approved this too early.  If something fails in `ansiMode`  finalResult is leaked, and because we throw in ansiMode on purpose at a minimum I would like to see this fixed with a closeOnExcept or something like that.